### ISSUE TITLE
Improve regrtest output

### ISSUE THREAD

**Issue by Yhg1s** [python/cpython#111474]
# Feature or enhancement

At the moment, the output of regrtest (what's run by `make test`) is a mixture of reports about test passes/fails/skips, running tests (if they run longer than a certain amount of time) and assorted test output, all interweaved:

```
[...]
0:00:42 load avg: 2.31 [ 52/469] test.test_multiprocessing_fork.test_processes passed
0:00:43 load avg: 2.31 [ 53/469] test.test_multiprocessing_forkserver.test_manager passed
0:00:53 load avg: 2.80 [ 54/469] test.test_concurrent_futures.test_wait passed (35.5 sec)
0:00:59 load avg: 3.38 [ 55/469] test.test_multiprocessing_spawn.test_misc passed -- running (1): test.test_multiprocessing_fork.test_threads 32.7 sec
0:00:59 load avg: 3.38 [ 56/469] test.test_multiprocessing_spawn.test_manager passed -- running (1): test.test_multiprocessing_fork.test_threads 33.4 sec
0:01:00 load avg: 3.38 [ 57/469] test___all__ passed -- running (1): test.test_multiprocessing_fork.test_threads 34.0 sec
0:01:00 load avg: 3.38 [ 58/469] test__locale passed -- running (1): test.test_multiprocessing_fork.test_threads 34.1 sec
0:01:00 load avg: 3.38 [ 59/469] test__opcode passed -- running (1): test.test_multiprocessing_fork.test_threads 34.2 sec
0:01:00 load avg: 3.38 [ 60/469] test__osx_support passed -- running (1): test.test_multiprocessing_fork.test_threads 34.3 sec
0:01:01 load avg: 3.38 [ 61/469] test__xxinterpchannels passed -- running (1): test.test_multiprocessing_fork.test_threads 35.2 sec
Traceback (most recent call last):
  File "<string>", line 2, in <module>
_xxinterpchannels.ChannelClosedError: channel 34 is closed
Traceback (most recent call last):
  File "<string>", line 2, in <module>
_xxinterpchannels.ChannelClosedError: channel 34 is closed
0:01:02 load avg: 3.38 [ 62/469] test.test_multiprocessing_forkserver.test_processes passed -- running (1): test.test_multiprocessing_fork.test_threads 36.1 sec
0:01:02 load avg: 3.38 [ 63/469] test__xxsubinterpreters passed -- running (1): test.test_multiprocessing_fork.test_threads 36.2 sec
Traceback (most recent call last):
  File "<string>", line 1, in <module>
SystemExit: 42
Traceback (most recent call last):
  File "<string>", line 1, in <module>
Exception: spam
  File "<string>", line 1
    print("spam"
         ^
SyntaxError: '(' was never closed
Traceback (most recent call last):
  File "<string>", line 3, in <module>
SystemExit
Traceback (most recent call last):
  File "<string>", line 3, in <module>
SystemExit: 42
0:01:02 load avg: 3.38 [ 64/469] test_abc passed -- running (1): test.test_multiprocessing_fork.test_threads 36.2 sec
0:01:02 load avg: 3.38 [ 65/469] test_abstract_numbers passed -- running (1): test.test_multiprocessing_fork.test_threads 36.3 sec
[...]
```

Personally, I find this much too hard to parse while the tests are running, especially when it starts reporting the long running tests. I would like to add an optional output mode for regrtest for use with vt100-capable terminals, using colors and multi-line redrawing to make the current state more visible. This requires some refactoring of regrtest internals to defer certain formatting decisions to the logger, rather than in the test task management itself, but it should be general enough that, for example, a Tkinter frontend could also be built.

### Linked PRs
* gh-111475

**Comment by vstinner** [comment]
regrtest was modified to add some colors. What's the status of this issue?

cc @hugovk
